src/python/module/smartredis/dataset_utils.py

```diff
@@ -131,7 +143,7 @@ def add_metadata_for_xarray(
 
     @staticmethod
     @_requires_xarray
-    def transform_to_xarray(dataset):
+    def transform_to_xarray(dataset: Dataset)->t.Dict:
```

Comment:
Am I right in assuming that this can be narrowed to
```py
def transform_to_xarray(dataset: Dataset) -> t.Dict[str, xr.DataArray]:
```
?

Reply:
Maybe. I think we'll be refining the type hints as we extend the code once Julia comes onboard